src/actions.ts

```diff
@@ -20,9 +20,24 @@ export const setValidation = (payload: {
   fieldName: string;
   validation: ValidationState;
   fromInitial?: boolean;
+  validationMessage?: any;
 }): SetValidationAction => {
   return {
-    payload,
+    payload: {
+      fieldName: payload.fieldName,
+      validation: payload.validationMessage
+        ? Object.entries(payload.validation).reduce<ValidationState>(
+            (acc, curr) => ({
```

Comment:
Ignore this if you feel it's unnecessary, but do these arguments need to have types? It may also improve readability, but I would say it's definitely optional

Reply:
Cheers, thankfully TS is clever enough to know what's going on here.

![Screenshot 2021-01-21 at 11 49 50](https://user-images.githubusercontent.com/6729304/105347452-d509f900-5bde-11eb-8f41-415c703ffcfa.png)